llama-index-integrations/vector_stores/llama-index-vector-stores-vertexaivectorsearch/llama_index/vector_stores/vertexaivectorsearch/base.py

```diff
@@ -0,0 +1,270 @@
+        endpoint_id (str): The fully qualified resource name of the created
+                           index endpoint in Vertex AI Vector Search.
+        gcs_bucket_name (Optional[str]):
+                           The location where the vectors will be stored for
+                           the index to be created in batch mode.
+        credentials_path (Optional[str]):
+                           The path of the Google credentials on the local file
+                           system.
+
+    Examples:
+        `pip install llama-index-vector-stores-vertexaivectorsearch`
+
+        ```python
+        from
+        vector_store = VertexAIVectorStore(
+            project_id=PROJECT_ID,
+            region=REGION,
+            index_id="<index_resource_name>"
+            endpoint_id="<index_endpoint_resource_name>"
+        )
+        ```
+    """
+
+    stores_text: bool = True
+    remove_text_from_metadata: bool = True
+    flat_metadata: bool = False
+
+    text_key: str
+
+    project_id: str
+    region: str
+    index_id: str
+    endpoint_id: str
+    gcs_bucket_name: str
+    credentials_path: Optional[str] = None
+
+    _index: MatchingEngineIndex = PrivateAttr()
+    _endpoint: MatchingEngineIndexEndpoint = PrivateAttr()
+    _index_metadata: dict = PrivateAttr()
+    _stream_update: bool = PrivateAttr()
+    _staging_bucket: storage.Bucket = PrivateAttr()
+    # _document_storage: GCSDocumentStorage = PrivateAttr()
+
+    def __init__(
+        self,
+        project_id: Optional[str] = None,
+        region: Optional[str] = None,
+        index_id: Optional[str] = None,
+        endpoint_id: Optional[str] = None,
+        gcs_bucket_name: Optional[str] = None,
+        credentials_path: Optional[str] = None,
+        text_key: str = DEFAULT_TEXT_KEY,
+        remove_text_from_metadata: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            project_id=project_id,
+            region=region,
+            index_id=index_id,
+            endpoint_id=endpoint_id,
+            gcs_bucket_name=gcs_bucket_name,
+            credentials_path=credentials_path,
+            text_key=text_key,
+            remove_text_from_metadata=remove_text_from_metadata,
+        )
+
+        """Initialize params."""
+        _sdk_manager = VectorSearchSDKManager(
+            project_id=project_id, region=region, credentials_path=credentials_path
+        )
+        self._index = _sdk_manager.get_index(index_id=index_id)
+        self._endpoint = _sdk_manager.get_endpoint(endpoint_id=endpoint_id)
+        self._index_metadata = self._index.to_dict()
+        self._stream_update = False
+        if self._index_metadata["indexUpdateMethod"] == "STREAM_UPDATE":
+            self._stream_update = True
+        self._staging_bucket = _sdk_manager.get_gcs_bucket(bucket_name=gcs_bucket_name)
+        # self._document_storage = GCSDocumentStorage(bucket=self._staging_bucket)
```

Comment:
I assume this is for doc+vector retrieval combined step. Should we remove this for the time being ?